lib/uksched/sched.c

```diff
@@ -442,3 +443,50 @@ UK_SYSCALL_R_DEFINE(int, sched_setaffinity, int, pid, long, cpusetsize,
 	UK_WARN_STUBBED();
 	return 0;
 }
+
+#define MEMBARRIER_SUPPORTED_CMDS (\
+	MEMBARRIER_CMD_GLOBAL | \
+	MEMBARRIER_CMD_GLOBAL_EXPEDITED | \
+	MEMBARRIER_CMD_REGISTER_GLOBAL_EXPEDITED | \
+	MEMBARRIER_CMD_PRIVATE_EXPEDITED | \
+	MEMBARRIER_CMD_REGISTER_PRIVATE_EXPEDITED | \
+	MEMBARRIER_CMD_PRIVATE_EXPEDITED_SYNC_CORE | \
+	MEMBARRIER_CMD_REGISTER_PRIVATE_EXPEDITED_SYNC_CORE)
+
+UK_LLSYSCALL_R_DEFINE(int, membarrier, int, cmd, unsigned int, flags,
```

Comment:
The `LL` registration variant will not provide the libc-wrapper function defintion if the libc relies on it (e.g., `lib/nolibc`). Is this intended because you added the libc-wrapper to `exportsyms.uk`.

Reply:
https://unikraft.org/docs/internals/syscall-shim

Reply:
I'm not sure I'd like to export `membarrier()` tbh, since the syscall signature differs from the libc one defined by musl, making it all a mess. Plus, nothing in unikraft core will ever need to rely on this.
I'll remove it from `exportsyms.uk`.